qiskit/quantum_info/operators/symplectic/pauli.py

```diff
@@ -0,0 +1,1006 @@
+        # Add extra phase from new Pauli to current
+        self._phase += value._phase
+
+    def delete(self, qubits):
+        """Return a Pauli with qubits deleted.
+
+        Args:
+            qubits (int or list): qubits to delete from Pauli.
+
+        Returns:
+            Pauli: the resulting Pauli with the specified qubits removed.
+
+        Raises:
+            QiskitError: if ind is out of bounds for the array size or
+                         number of qubits.
+        """
+        if isinstance(qubits, (int, np.int)):
+            qubits = [qubits]
+        if max(qubits) > self.num_qubits - 1:
+            raise QiskitError(
+                "Qubit index is larger than the number of qubits "
+                "({}>{}).".format(max(qubits), self.num_qubits - 1))
+        if len(qubits) == self.num_qubits:
+            raise QiskitError("Cannot delete all qubits of Pauli")
+        z = np.delete(self._z, qubits, axis=1)
+        x = np.delete(self._x, qubits, axis=1)
+        return Pauli(z, x, phase=self.phase)
+
+    def insert(self, qubits, value):
+        """Insert a Pauli at specific qubit value.
+
+        Args:
+            qubits (int or list): qubits index to insert at.
+            value (Pauli): value to insert.
+
+        Returns:
+            Pauli: the resulting Pauli with the entries inserted.
+
+        Raises:
+            QiskitError: if the insertion qubits are invalid.
+        """
+        if not isinstance(value, Pauli):
+            value = Pauli(value)
+
+        # Initialize empty operator
+        ret_qubits = self.num_qubits + value.num_qubits
+        ret = Pauli(np.zeros(ret_qubits, dtype=np.bool),
+                    np.zeros(ret_qubits, dtype=np.bool))
+        if isinstance(qubits, (int, np.int)):
+            if value.num_qubits == 1:
+                qubits = [qubits]
+            else:
+                qubits = list(range(qubits, qubits + value.num_qubits))
+        if len(qubits) != value.num_qubits:
+            raise QiskitError(
+                "Number of indices does not match number of qubits for "
+                "the inserted Pauli ({}!={})".format(len(qubits),
+                                                     value.num_qubits))
+        if max(qubits) > ret.num_qubits - 1:
+            raise QiskitError(
+                "Index is too larger for combined Pauli number of qubits "
+                "({}>{}).".format(max(qubits), ret.num_qubits - 1))
+        # Qubit positions for original op
+        self_qubits = [i for i in range(ret.num_qubits) if i not in qubits]
+        ret[self_qubits] = self
+        ret[qubits] = value
+        return ret
+
+    # ---------------------------------------------------------------------
+    # Representation conversions
+    # ---------------------------------------------------------------------
+
+    def __hash__(self):
+        """Make hashable based on string representation."""
+        return hash(self.__str__())
```

Comment:
This occurs collisions of long paulis due to truncation of `str`, e.g.,
```python
In [28]: hash(Pauli('X' * 200))
Out[28]: -7113259572996037805

In [29]: hash(Pauli('X' * 300))
Out[29]: -7113259572996037805
```
It would be good if we can avoid such collisions.

Reply:
Good catch. I forgot to update this function after adding truncation to string